gensim/models/nmf.py

```diff
@@ -606,7 +606,7 @@ def update(self, corpus, chunksize=None, passes=None, eval_every=None):
                         chunk,
                         num_terms=self.num_tokens,
                         num_docs=len(chunk),
-                        num_nnz=sum(1 for _ in itertools.chain.from_iterable(chunk)),
+                        num_nnz=sum(len(doc) for doc in chunk),
```

Comment:
Is `doc` always guaranteed to have `len()`?

Reply:
Hm. Ok, we `doc` isn't required to have `len`. Explicit specification of the parameters speed up the conversion a bit, but I think it's not that much of a gain. I'll get back an implicit conversion then.